hack/debug-network.sh

```diff
@@ -0,0 +1,729 @@
+    local base_pod_name=$1
+    local base_pod_addr=$2
+    local base_pod_pid=$3
+    local base_pod_port=$4
+    local base_pod_vnid=$5
+    local base_pod_ether=$6
+
+    echo "EXTERNAL TRAFFIC:"
+    echo ""
+    echo "${base_pod_name} -> example.com"
+    # This address is from a range which is reserved for documentation examples
+    # (RFC 5737) and not allowed to be used in private networks, so it should be
+    # guaranteed to only match the default route.
+    echo_and_eval ovs-appctl ofproto/trace br0 "in_port=${base_pod_port},reg0=${base_pod_vnid},ip,nw_src=${base_pod_addr},nw_dst=198.51.100.1"
+    echo ""
+    echo "example.com -> ${base_pod_name}"
+    echo_and_eval ovs-appctl ofproto/trace br0 "in_port=2,ip,nw_src=198.51.100.1,nw_dst=${base_pod_addr},dl_dst=${base_pod_ether}"
+    echo ""
+
+    if nsenter -n -t "${base_pod_pid}" -- ping -c 1 -W 2 "www.google.com"  &> /dev/null; then
+        echo "ping www.google.com -> success"
+    else
+        echo "ping www.google.com -> failed"
+    fi
+}
+
+function do_pod_service_connectivity_check() {
+    local namespace=$1
+    local base_pod_name=$2
+    local base_pod_addr=$3
+    local base_pod_pid=$4
+    local base_pod_port=$5
+    local base_pod_vnid=$6
+    local base_pod_ether=$7
+    local service_name=$8
+    local service_addr=$9
+    local service_proto=${10}
+    local service_port=${11}
+
+    echo "service, ${namespace} namespace:" | tr '[a-z]' '[A-Z]'
+    echo ""
+
+    echo "${base_pod_name} -> ${service_name}"
+    echo_and_eval ovs-appctl ofproto/trace br0 "in_port=${base_pod_port},reg0=${base_pod_vnid},${service_proto},nw_src=${base_pod_addr},nw_dst=${service_addr},${service_proto}_dst=${service_port}"
+    echo ""
+    echo "${service_name} -> ${base_pod_name}"
+    echo_and_eval ovs-appctl ofproto/trace br0 "in_port=2,${service_proto},nw_src=${service_addr},nw_dst=${base_pod_addr},dl_dst=${base_pod_ether}"
+    echo ""
+
+    # In bash, redirecting to /dev/tcp/HOST/PORT or /dev/udp/HOST/PORT opens a connection
+    # to that HOST:PORT. Use this to test connectivity to the service; we can't use ping
+    # like in the pod connectivity check because only connections to the correct port
+    # get redirected by the iptables rules.
+    if nsenter -n -t "${base_pod_pid}" -- timeout 1 bash -c "echo -n '' > /dev/${service_proto}/${service_addr}/${service_port} 2>/dev/null"; then
+        echo "connect ${service_addr}:${service_port} -> success"
+    else
+        echo "connect ${service_addr}:${service_port} -> failed"
+    fi
+
+    echo ""
+    echo ""
+}
+
+function get_config_path_from_service() {
+    # 'node', 'master', 'master api', 'master controllers'
+    local service_type=$1
+    # 'atomic-openshift-node.service', 'atomic-openshift-master-controllers.service', etc
+    local service_name=$2
+    local config varfile
+
+    config="$(ps wwaux | grep -v grep | sed -ne "s/.*openshift start ${service_type} --.*config=\([^ ]*\.yaml\).*/\1/p")"
```

Comment:
as we know we're not on OSX (or if we don't, your `sed` callouts are somewhat broken) use `pgrep` instead of `ps | grep`